src/infrastructure/TransactionSearchCriteria.ts

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2020 NEM
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { TransactionType } from '../model/transaction/TransactionType';
+import { QueryParams } from './QueryParams';
+
+ /**
+  * Transaction search criteria for filtering transaction list
+  */
+export class TransactionSearchCriteria extends QueryParams {
+    /**
+     * Transaction type filter
+     */
+    public type?: string;
```

Comment:
This should be a list of enums so the model is cleaner. Then the repository should do the csv to the param which is an implementation detail (I think there should be a way in open api to not use csv params)

Reply:
I decided to remove this class instead using the existing queryParam class as implemented the setter

Reply:
The problem with QuerayParams is that it's used for different queries that may not have a transaction type filtering.

For example:

https://github.com/nemtech/nem2-sdk-typescript-javascript/blob/master/src/infrastructure/NamespaceRepository.ts#L63

Now they accept transaction types that aren't used

The plan is you have common pagination/sorting param in a top-level class (QueryParams although I would call it SearchCriteria) and then you have a subclass adding the custom filters (like TransactionSearchCriteria). Hopely one subclass per entity type (Account, Transaction, Namespace)

I don't like the name QueryParams. It is implementation-specific. What happens if tomorrow we send the params as a post body or we call the database directly like the e2e repositories? 

The search and pagination will have a big refactoring soon, this can be revisited then.

@dgarcia360 @evias ?

Reply:
Replaced by TransactionFilter class which acts now as a separate parameter on account http